### PR TITLE
Add L12 to priors

### PR DESCRIPTION
**Pull request by Nils-Laurent**
In this PR, I propose to add the L12Prior to deepinv. I also propose a fix in TVPrior, when evaluatin `g`, because when batch_size > 1, the shape is not correct.

### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: deepinv/tests/test_optim.py [hunk 1]
```diff
@@ -196,6 +196,58 @@ def A_adjoint(v):
...
+
+    physics = dinv.physics.Blur(
+        dinv.physics.blur.gaussian_blur(sigma=(0.7, 0.7), angle=0.0),
+        padding="circular",
+        device=device,
+        noise_model=dinv.physics.noise.GaussianNoise(sigma=0.01),
+    )
+
+    y = physics(x)
+    max_iter = 1000
```
Comment (tachella) [inline deepinv/tests/test_optim.py:221]: Hi @Nils-Laurent, did you find a way to reduce the number of iterations in the test?

Comment (Nils-Laurent) [reply deepinv/tests/test_optim.py:221]: Hi @tachella, I created this test because it requires a low number of iteration to converge. I noticed that L12 in test_priors_algo triggered the assert because the L2 axis was wrong, when I set it to the channels' axis, it passes the test

Comment (Nils-Laurent) [reply deepinv/tests/test_optim.py:221]: Maybe I should simply use L12Prior in test_priors_algo, that would make the PR more concise

**Review by samuro95 — COMMENTED** [review]
Thanks for this contribution, and for correcting the TV prior.  I have no additional comments.

File: deepinv/optim/prior.py [hunk 2]
```diff
@@ -657,3 +658,68 @@ def forward(self, x):
...
+    r"""
+    :math:`\ell_{1,2}` prior :math:`\reg{x} = \sum_i\| x_i \|_2`.
+
+    |sep|
+
+    :Examples:
+    >>> import torch
+    >>> from deepinv.optim import L12Prior
+    >>> seed = torch.manual_seed(0) # Random seed for reproducibility
+    >>> x = torch.randn(1, 1, 3, 3) # Define random 3x3 image
```
[the remark this answers is not in the document]
Comment (Nils-Laurent) [reply deepinv/optim/prior.py]: I updated the example with 2 batches and I verify that output prox and g is consistent when computing it on elements of the batch separately.

**Review by tachella — APPROVED** [review]
